cmd/query/app/server_test.go

```diff
@@ -616,14 +616,14 @@ func TestServerSinglePort(t *testing.T) {
 		jtracer.NoOp())
 	require.NoError(t, err)
 	require.NoError(t, server.Start())
-	t.Cleanup(func() {
+	defer func() {
 		require.NoError(t, server.Close())
-	})
+	}()
```

Comment:
if my understanding is correct,  t.Cleanup also handles cleanup after the test, using defer within the function makes the order of operations clearer and more controllable, which prevents race around conditions

Reply:
I don't see how it prevents race conditions. Cleanups are also run in deterministic last-to-first order, same as defer's. I don't think the order of cleanups is the root cause of this issue.

Reply:
Ok let me look into it further

Reply:
Right now i have added a logic that , it will try to close the server only once, even if it is called twice.
If this also will not work, Should i look into checking the port if it is bound and then trying to close the session

Reply:
@yurishkuro : can you pls have a look at this change

Reply:
I don't believe these changed address anything. The test is already doing everything right in terms of closing. I think the issue is in the actual code of the servers.

Reply:
I've been troubleshooting a persistent timeout issue with the TestServerSinglePort test in the server_test.go Here's what I've done so far:

1. Dynamic Port Assignment: Used ":0" for automatic port selection.
2. Enhanced Logging: Improved logging in TestServerSinglePort, Start, and Close methods.
3. Error Handling: Ensured graceful handling and logging of server start/stop errors.

Despite these changes, the test still times out after 5 minutes. It seems the issue might be within the server code itself. Any guidance would be greatly appreciated.